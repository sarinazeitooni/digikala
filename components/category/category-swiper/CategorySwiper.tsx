import React from 'react';
import { Swiper, SwiperSlide } from 'swiper/react';
import data from "../../specials/texts/data";
import CategoryCard from "../category-card/CategoryCard";
import { v4 as uuidv4 } from 'uuid';
import useViewPort from "../../../customIsMobileHook/useViewPort";
const CategorySwiper = ()=>{
    const isMobile = useViewPort();

    return(
        <Swiper style={{ '--swiper-navigation-color': "black" , '--swiper-navigation-size' : '26px'}} slidesPerView={isMobile ? 2 :6} navigation>
                {data.cards.map((item : {hasDiscount : boolean , discount : number , img : string , price : number , title : string })=>{
                    return(
                        <SwiperSlide key={uuidv4()}>
                            <CategoryCard hasDiscount = {item.hasDiscount} discount={item.discount} img={item.img} price={item.price} title={item.title}/>
                        </SwiperSlide>
                    )
                })}
            </Swiper>
    )
}
export default CategorySwiper;