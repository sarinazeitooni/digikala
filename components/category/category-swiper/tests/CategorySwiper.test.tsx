import {shallow} from "enzyme";
import CategorySwiper from "../CategorySwiper";
describe("CategorySwiper component" , ()=>{
    it("should render correctly" , ()=>{
        const container = shallow(<CategorySwiper/>);
        expect(container).toMatchSnapshot();
    })
});