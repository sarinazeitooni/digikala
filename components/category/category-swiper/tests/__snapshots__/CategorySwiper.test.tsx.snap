// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySwiper component should render correctly 1`] = `ShallowWrapper {}`;
